audit_log.py

```diff
@@ -0,0 +1,170 @@
+import socket
+import ssl
+import sys
+import argparse
+import ConfigParser
+import requests
+from jinja2 import Template
+import os
+import json
+import time
+import logging
+import logging.handlers
+from cb_defense_syslog import *
+import traceback
+import hashlib
+import fcntl
+
+
+#Get Audit Logs will return response
+def get_audit_logs(url, api_key_query, api_connector_id_query, ssl_verify, logger, proxies=None):
```

Comment:
You don't need to pass in logger. Common practice in python is to create that in each file. python will execute any code outside of def or class when imported or referenced.

Reply:
You can use CBAPI for a reference https://github.com/carbonblack/cbapi-python/blob/master/src/cbapi/psc/threathunter/models.py#L9

Reply:
I totally agree that sending in a logger through each function is not the best way to do it and went back and did some modifications. I used the following approach to fix the problem: https://docs.python.org/2.7/howto/logging.html#logging-from-multiple-modules. I think creating the logger in the main file and then importing it throughout the rest of the program would be a better approach, rather than creating a new one in each script. Let me know if you have any other questions. The updates will be in the next push. Thanks!

Reply:
Since this is a smaller application, I can see the benefit to using that mechanism. Though it also mentions the drawback which obfuscates the location where the log was generated. So we may want to consider this section https://docs.python.org/3.7/howto/logging.html#logging-advanced-tutorial for larger applications

Reply:
The goal is mult-module logging, so that the instantiation of the logger is at the top scope within the service. The advanced section was giving a better way to inject the module into the logger itself.

Reply:
I found a logging bug in my code and fixed it. Will be fixed in the next  push

Reply:
It was a bug with the new approaches not with the original scripts